util/Ozone_Monitor/nwprod/oznmon_shared/sorc/oznmon_time.fd/oznmon_read_diag.f90

```diff
@@ -698,14 +698,30 @@ subroutine read_ozndiag_data_nc( ftin, header_fix, data_fix, data_nlev, data_ext
           write(6,*) 'WARNING:  unable to read global var Time from file '
        end if
 
-       !-----------------------------------------------
+       !-------------------------------------------------------
        ! lat, lon, obstime are dimensioned to nrecords
-       ! read those as nobs * nlevs
+       ! read those as nobs * nlevs with all lat/lon values 
+       ! for a given obs being the same.  So a source with 22
+       ! levels has that value repeated 22 times for each 
+       ! obs.  The data_fix structure is dimentioned to nobs,
+       ! not nrecords to match the binary file structure.
+       !
+       ! Note that only sbuv2_n19 and ompsnp_npp (at present)
+       ! have nlevs > 1.
        !
        do ii=1,ntobs
-          data_fix(ii)%lat     =     lat(ii + ((ii-1)*nlevs) )
-          data_fix(ii)%lon     =     lon(ii + ((ii-1)*nlevs) )
-          data_fix(ii)%obstime = obstime(ii + ((ii-1)*nlevs) )
+          if( nlevs > 1 ) then
+             data_fix(ii)%lat     =     lat(1 + ((ii-1)*nlevs) )
+             data_fix(ii)%lon     =     lon(1 + ((ii-1)*nlevs) )
+             data_fix(ii)%obstime = obstime(1 + ((ii-1)*nlevs) )
+          else
+             data_fix(ii)%lat     =     lat(ii)
+             data_fix(ii)%lon     =     lon(ii)
+             data_fix(ii)%obstime = obstime(ii)
+          end if
+          
+          print *, 'ii, data_fix(ii)%lat, data_fix(ii)%lon = ', ii, data_fix(ii)%lat, data_fix(ii)%lon
```

Comment:
Same comment as above.   Should we comment out this diagnostic print?

Reply:
Change has been made.